otx/core/data/adapter/detection_dataset_adapter.py

```diff
@@ -36,16 +36,18 @@ def get_otx_dataset(self) -> DatasetEntity:
                     image = Image(file_path=datumaro_item.media.path)
                     shapes = []
                     for ann in datumaro_item.annotations:
-                        if self.task_type is TaskType.DETECTION and ann.type == AnnotationType.polygon:
+                        if (
+                            str(self.task_type) is TaskType.INSTANCE_SEGMENTATION.name
```

Comment:
Why do we need `str(*)` and `*.name`?

Reply:
Can't we just use `self.task_type is TaskType.INSTANCE_SEGMENTATION`?

Reply:
This part was corrected after discussing with sungman.
https://github.com/openvinotoolkit/training_extensions/pull/1572/commits/19c016e8cd4d03a38bb489e2d933c0724f108ebe